modules/core/src/main/java/org/apache/ignite/internal/processors/cache/verify/ViewCacheClosure.java

```diff
@@ -128,13 +144,27 @@ public ViewCacheClosure(String regex, VisorViewCacheCmd cmd) {
                     ci.setAtomicityMode(desc.cacheConfiguration().getAtomicityMode());
                     ci.setMapped(mapped(desc.cacheName()));
 
+                    GridCacheProcessor cacheProcessor = k.context().cache();
+                    ci.setOffHeapPrimaryEntriesCnt(getOffHeapCountFromCache(cacheProcessor, ci.getCacheName()));
+
                     cacheInfo.add(ci);
                 }
 
                 return cacheInfo;
         }
     }
 
+    /**
+     * Method for getting cache size via GridCacheProcessor
+     * @param cacheProcessor - GridCacheProcessor
+     * @param cacheName - name
+     * @return off-heap cache size
+     */
+    private Long getOffHeapCountFromCache(GridCacheProcessor cacheProcessor, String cacheName) throws Exception {
+        IgniteCache<Object, Object> cache = cacheProcessor.jcache(cacheName);
+        return cache.sizeLong(CachePeekMode.OFFHEAP);
```

Comment:
why we need OFFHEAP param here ? just calling cache.sizeLong() would be enough i hope.